internal/httpserve/bankid.go

```diff
@@ -3,246 +3,178 @@ package httpserve
 )
 
 func RegisterBankIDServer(e *echo.Echo, client *bankid.API) {
-	e.POST("/bankid/v6/auth", func(e echo.Context) error {
-		b, err := io.ReadAll(e.Request().Body)
+	e.POST("/bankid/v6/auth", func(c echo.Context) error {
+		b, err := io.ReadAll(c.Request().Body)
 		if err != nil {
-			return e.JSON(400, bankid.GenericResponse{Message: "invalid request payload"})
+			return c.JSON(400, bankid.GenericResponse{Message: "invalid request payload"})
 		}
 
 		var request bankid.AuthSignRequest
 		err = json.Unmarshal(b, &request)
 		if err != nil {
 			fmt.Printf("ERR: failed to unmarshal auth request message: %s\n", err.Error())
-			return e.JSON(400, bankid.GenericResponse{Message: "invalid request payload content"})
+			return c.JSON(400, bankid.GenericResponse{Message: "invalid request payload content"})
 		}
 
-		res, err := client.Auth(e.Request().Context(), &request)
+		res, err := client.Auth(c.Request().Context(), &request)
 		if err != nil {
 			fmt.Printf("ERR: initiating auth request against bankid: %s\n", err.Error())
-			return e.JSON(500, "failed to initiate auth against BankId")
+			return c.JSON(500, "failed to initiate auth against BankId")
 		}
 
 		// In the case a client wants to initiate a new request every second instead of relying on SSE
 		// we respond with the first entry and then close the connection
-		response := e.QueryParam("type")
+		response := c.QueryParam("type")
 		if response == "once" {
-			msg := bankid.AuthSignAPIResponse{
-				OrderRef: res.OrderRef,
-				URI:      fmt.Sprintf("bankid:///?autostarttoken=%s&redirect=null", res.AutoStartToken),
-				QR:       res.BuildQrCode(0),
-			}
-
-			return e.JSON(200, msg)
+			return c.JSON(200, res.APIResponse(0))
 		}
 
-		sender, err := sse.NewSender(e.Response().Writer)
+		sender, interrupt, err := client.WatchForChange(c, res.OrderRef)
 		if err != nil {
-			fmt.Printf("ERR: failed to setup auth response stream: %s\n", err.Error())
-			return e.JSON(500, "failed to setup response stream")
+			return c.JSON(500, "failed to setup response stream")
 		}
 
-		sender.Prepare()
-
-		interrupt := client.WatchForChange(e.Request().Context(), res.OrderRef)
-
 		for i := 0; i < 30; i++ {
-			select {
-			case <-interrupt:
-				return e.JSON(http.StatusOK, bankid.Empty{})
-			default:
-				break
-			}
-
-			msg := bankid.AuthSignAPIResponse{
-				OrderRef: res.OrderRef,
-				URI:      fmt.Sprintf("bankid:///?autostarttoken=%s&redirect=null", res.AutoStartToken),
-				QR:       res.BuildQrCode(i),
-			}
-
-			var bytes []byte
-			bytes, err = json.Marshal(msg)
-			if err != nil {
-				fmt.Printf("ERR: failed to build auth response message: %s\n", err.Error())
-				return e.JSON(500, "failed to build response message")
-			}
-
-			event := sse.Event{
-				Id:    strconv.Itoa(i),
-				Event: "message",
-				Data:  string(bytes),
-			}
-
-			err = sender.Send(event)
+			err = sender.Send("message", res.APIResponse(i)) // TODO: Should we change "message" to something else?
```

Comment:
When we have received a pending, we should probably stop sending QR codes since the code has been scanned